### PR TITLE
Update spotbugs and checkstyle

### PR DESCRIPTION
**Pull request by sknot-rh**
Signed-off-by: Stanislav Knot <sknot@redhat.com>

### Type of change
- Chores

### Description
Updated spotbugs and checkstyle. Spotbugs now checks access to the private members from the static methods. So using singleton pattern triggers `*_EXPOSE_REP*` warnings (also public method getting or setting private field). I added those to spotbugs exclude.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperatorTest.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ public void testCreateWhenExistsWithChangeIsAPatch(VertxTestContext context) {
         HasMetadata hasMetadata = mock(HasMetadata.class);
         when(mockResource.get()).thenReturn(resource);
 
-        when(mockResource.withPropagationPolicy(true ? DeletionPropagation.FOREGROUND : DeletionPropagation.ORPHAN)).thenReturn(mockR);
+        when(mockResource.withPropagationPolicy(DeletionPropagation.FOREGROUND)).thenReturn(mockR);
```
Comment (scholzj) [inline operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperatorTest.java:103]: Why change this?

File: operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperatorTest.java [hunk 2]
```diff
@@ -132,7 +132,7 @@ public void testCreateWhenExistsWithoutChangeIsNotAPatch(VertxTestContext contex
         T resource = resource();
         Resource mockResource = mock(resourceType());
         when(mockResource.get()).thenReturn(resource);
-        when(mockResource.withPropagationPolicy(true ? DeletionPropagation.FOREGROUND : DeletionPropagation.ORPHAN)).thenReturn(mockResource);
+        when(mockResource.withPropagationPolicy(DeletionPropagation.FOREGROUND)).thenReturn(mockResource);
```
Comment (scholzj) [inline operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperatorTest.java:135]: Same as above.

File: systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/VerifiableClient.java [hunk 3]
```diff
@@ -122,8 +122,9 @@ public VerifiableClient(VerifiableClientBuilder verifiableClientBuilder) {
 
         if (clientType == ClientType.CLI_KAFKA_VERIFIABLE_CONSUMER) {
             this.consumerGroupName = verifiableClientBuilder.consumerGroupName;
-            this.clientArgumentMap.put(ClientArgument.GROUP_ID, consumerGroupName);
             this.consumerInstanceId = verifiableClientBuilder.consumerInstanceId;
+            this.clientArgumentMap.put(ClientArgument.GROUP_ID, consumerGroupName);
+            this.clientArgumentMap.put(ClientArgument.GROUP_INSTANCE_ID, consumerInstanceId);
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/VerifiableClient.java:127]: Why this change? Seems unrelated to spotbugs or checkstyle.

File: systemtest/src/main/java/io/strimzi/systemtest/logs/TestExecutionWatcher.java [hunk 4]
```diff
@@ -20,6 +20,8 @@
 
 public class TestExecutionWatcher implements TestExecutionExceptionHandler, LifecycleMethodExecutionExceptionHandler {
 
+    private static final Random RANDOM = new Random();
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/logs/TestExecutionWatcher.java:23]: Why do you use once static field and once non-static? Shouldn't it use everywhere the same?

**Review by scholzj — COMMENTED** [review]
> Spotbugs now checks access to the private members from the static methods. So using singleton pattern triggers *_EXPOSE_REP* warnings. I added those to spotbugs exclude.

Perhaps you can elaborate some more on this and provide some examples of what it doesn't like and why should it be excluded?

File: operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperatorTest.java (hunk 1 above)
Comment (sknot-rh) [reply operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperatorTest.java:103]: Well, `true ? DeletionPropagation.FOREGROUND : DeletionPropagation.ORPHAN` is equal to 
```
if (true) {
    doSomething();`
    } else {
    doSomethingElse();
}
``` 
so it never gets to the else branch, does it?

Comment (scholzj) [reply operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperatorTest.java:103]: Good point. I didn't noticed it was just `true` and not some actual condition. Weird code :-o.

File: systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/VerifiableClient.java (hunk 3 above)
Comment (sknot-rh) [reply systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/VerifiableClient.java:127]: Spotbugs reported `consumerGroupName` as unused. Looking at the code I thing it might a bug.
It was partially fixed in https://github.com/strimzi/strimzi-kafka-operator/pull/6156/files#diff-f93d2fac0903e6c67c6a2111f4b65fa4b61a2fd27690459d80d0ae4c6561a70eR126
@fvaleri @see-quick Could you please take a look?

**Review by scholzj — APPROVED** [review]

File: systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/VerifiableClient.java (hunk 3 above)
Comment (fvaleri) [reply systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/VerifiableClient.java:128]: Yeah, that actually revealed a bug and I forgot to add it to the arguments map. Thanks.
